src/main/java/uk/gov/companieshouse/pscdataapi/service/CompanyPscService.java

```diff
@@ -0,0 +1,98 @@
+
+    private static final String APPLICATION_NAME_SPACE = "psc-data-api";
+    private static final Logger logger = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    private final DateTimeFormatter dateTimeFormatter =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+
+    @Autowired
+    private CompanyPscTransformer transformer;
+    @Autowired
+    private CompanyPscRepository repository;
+
+    /**
+     * Save or update a natural disqualification.
+     * @param contextId     Id used for chsKafkaCall.
+     * @param requestBody   Data to be saved.
+     */
+    public void insertPscRecord(String contextId, FullRecordCompanyPSCApi requestBody) {
+        String notificationId = requestBody.getExternalData().getNotificationId();
+        boolean isLatestRecord = isLatestRecord(notificationId, requestBody
+                .getInternalData().getDeltaAt());
+        if (isLatestRecord) {
+            PscDocument document = transformer.transformPsc(notificationId, requestBody);
+            save(contextId, notificationId, document);
+        } else {
+            logger.info("PSC not persisted as the record provided is not the latest record.");
+        }
+    }
+
+    private boolean isLatestRecord(String notificationId, OffsetDateTime deltaAt) {
+        String formattedDate = deltaAt.format(dateTimeFormatter);
+        List<PscDocument> pscDocuments = repository
+                .findUpdatedPsc(notificationId, formattedDate);
+        return pscDocuments.isEmpty();
+    }
+
+    /**
+     * Save or update the mongo record.
+     * @param contextId Chs kafka id.
+     * @param notificationId Mongo id.
+     * @param document  Transformed Data.
+     */
+    private void save(String contextId, String notificationId,
+                      PscDocument document) {
+        Created created = getCreatedFromCurrentRecord(notificationId);
+        if (created == null) {
+            document.setCreated(new Created().setAt(LocalDateTime.now()));
+        } else {
+            document.setCreated(created);
+        }
+
+        try {
+            repository.save(document);
+            logger.info(String.format(
+                    "Company PSC record is updated in MongoDb for context id: %s and id: %s",
+                    contextId, notificationId));
+        } catch (IllegalArgumentException illegalArgumentEx) {
+            throw new BadRequestException(illegalArgumentEx.getMessage());
+        }
+    }
+
+    /**
+     * Check whether we are updating a record and if so persist created time.
+     * @param notificationId Mongo Id.
+     * @return created if this is an update save the previous created to the new document.
+     */
+    private Created getCreatedFromCurrentRecord(String notificationId) {
+        Created created = new Created();
+        try {
+            Optional<PscDocument> doc = repository.findById(notificationId);
+            return doc.map(PscDocument::getCreated).orElse(null);
```

Comment:
This is nice, could also do it in one line
```suggestion
            return repository.findById(notificationId)
                    .map(PscDocument::getCreated).orElse(null);
```